### ISSUE TITLE
Warning mismatches on tensorflow macos

### ISSUE THREAD

**Issue by stellaraccident** [openxla/iree#4489]
```
2021-01-12T11:06:17.7726980Z [1mexternal/org_tensorflow/tensorflow/compiler/mlir/tensorflow/ir/tf_traits.h:91:20: [0m[0;1;31merror: [0m[1mloop variable 'operand_type' is always a copy because the range of type 'mlir::Operation::operand_type_range' (aka 'ValueTypeRange<mlir::OperandRange>') does not return a reference [-Werror,-Wrange-loop-analysis][0m
2021-01-12T11:06:17.7729510Z   for (const auto& operand_type : op->getOperandTypes()) {
2021-01-12T11:06:17.7735430Z [0;1;32m                   ^
2021-01-12T11:06:17.7737130Z [0m[1mexternal/org_tensorflow/tensorflow/compiler/mlir/tensorflow/ir/tf_traits.h:91:8: [0m[0;1;30mnote: [0muse non-reference type 'mlir::Type'[0m
2021-01-12T11:06:17.7738980Z   for (const auto& operand_type : op->getOperandTypes()) {
2021-01-12T11:06:17.7740020Z [0;1;32m       ^~~~~~~~~~~~~~~~~~~~~~~~~~
2021-01-12T11:06:17.7740760Z [0m2 errors generated.
```

```
2021-01-12T11:06:17.7716040Z [1mexternal/org_tensorflow/tensorflow/core/framework/type_index.h:61:17: [0m[0;1;31merror: [0m[1munused variable 'hash_bit' [-Werror,-Wunused-variable][0m
2021-01-12T11:06:17.7717460Z     static bool hash_bit[1];
2021-01-12T11:06:17.7718690Z [0;1;32m             
```

I am considering just not making warnings errors for tensorflow on macos. I just don't think there is a lot of value in traipsing down these issues in someone else's project.

**Comment by GMNGeoffrey** [comment]
I think this would be fixed by #9306 as well

**Comment by ScottTodd** [comment]
Obsolete - we no longer build TensorFlow from C++ sources.
